### PR TITLE
Github Action Linter (YAML)

### LINKED ISSUE #21207: CI: Move Linter (YAML) to Github Action
#### Is your feature request related to a problem?

That permits to optimize time on Travis CI

#### Describe the solution you'd like

Move linter to a Github Action and remove it from Travis CI

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Move linter from Travis CI to Github Action
| Type?         | improvement
| Category?     | PM
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #21207
| How to test?  | Travis is :green_circle: & GA is :green_circle: 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21479)

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Linter
+on: [push, pull_request]
+jobs:
+  yaml:
+    runs-on: ubuntu-latest
+    name: YAML Linter
+    steps:
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v1
```
Comment (atomiix) [inline .github/workflows/lint.yml]:
```suggestion
        uses: shivammathur/setup-php@v2
```

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          php-version: '7.3'
+          extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv
+      - uses: actions/checkout@v1
```
Comment (atomiix) [inline .github/workflows/lint.yml]:
```suggestion
      - uses: actions/checkout@v2
```

**Review by atomiix — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]

File: src/PrestaShopBundle/Resources/config/services/adapter/product.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 services:
-    _defaults:
+      _defaults:
```
Comment (kpodemski) [inline src/PrestaShopBundle/Resources/config/services/adapter/product.yml]: 🤔

**Comment by kpodemski** [comment]
Travis is not happy with one file, and this file is after YAML Linter? 🤔

**Comment by matks** [comment]
> Travis is not happy with one file, and this file is after YAML Linter? 🤔

I guess the YAML Linter wants the file to be this way, but this is considered invalid YAML by Symfony so Symfony cannot boot because of invalid config file.

So Symfony ⚔️ YAML Linter 😆

File: src/PrestaShopBundle/Resources/config/services/adapter/product.yml (hunk 3 above)
Comment (Progi1984) [reply src/PrestaShopBundle/Resources/config/services/adapter/product.yml]: (I wanted to break the CI :p But YamlLint seems to be hard than `php bin/console yaml:lint`).

**Review by jolelievre — DISMISSED** [review]
Big ❤️.  on this!!

Just a question, which command should we run to auto-fix when required? Should it be documented in a Readme? Or maybe the CI indicates which command to run when it fails?

**Comment by jolelievre** [comment]
Oups seems like the CI is not happy actually ^^

File: .github/workflows/yamllint/.yamllint.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    braces:
+        min-spaces-inside: 1
+        max-spaces-inside: 1
+    brackets:
+        min-spaces-inside: 1
+        max-spaces-inside: 1
+    document-start: disable
+    indentation:
+        ## Check .editorconfig
+        spaces: 4
```
Comment (PierreRambaud) [inline .github/workflows/yamllint/.yamllint.yml]:
```suggestion
        spaces: 2
```

Prefer 2as we do for html and json for the moment :D

Comment (PierreRambaud) [reply .github/workflows/yamllint/.yamllint.yml]: You can read more here https://docs.saltproject.io/en/3000/topics/troubleshooting/yaml_idiosyncrasies.html#:%7E:text=Indentation,uniform%20and%20not%20deeply%20nested.

Comment (Progi1984) [reply .github/workflows/yamllint/.yamllint.yml]: Bad joke... @eternoendless define the inverse thing : #20324 since the 1.7.7.0.

Comment (PierreRambaud) [reply .github/workflows/yamllint/.yamllint.yml]: We can change the editorconfig file :sweat_smile:

Comment (PierreRambaud) [reply .github/workflows/yamllint/.yamllint.yml]: We also say it in our documentation: https://devdocs.prestashop.com/1.7/themes/getting-started/guidelines/#general

> 2 spaces for all other file types.

File: .github/workflows/lint.yml [hunk 5]
```diff
@@ -85,7 +84,29 @@ jobs:
 
       -   name: Run Lint Yaml on `.t9n.yml`
           run: php bin/console lint:yaml .t9n.yml
+  yamllint:
```
Comment (jolelievre) [inline .github/workflows/lint.yml:87]: I know the lint is not run on these github workflow files, but seems like they mix 2 an 4 spaces tabulation 😅
But it's not mandatory for this PR, maybe they can be added later just to have the same rules everywhere

**Review by jolelievre — COMMENTED** [review]
Thank you @Progi1984 this will be so much cleaner
Just two small issues:
- there are conflicting files, you'll need a rebase
- there are warnings introduced by the Github action in `app/config/config.yml` are they blocking errors or just warnings that can be ignored?

**Comment by Progi1984** [comment]
@jolelievre Rebase done & YAMLLint applied on .github directory.

Warnings are not blocking.

**Review by jolelievre — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @Progi1984

**Pull request merged**